lib/vdsm/virt/vm.py

```diff
@@ -4443,6 +4443,13 @@ def diskReplicateFinish(self, srcDisk, dstDisk):
                            "srcDisk: %r, job: %r)",
                            drive.name, srcDisk, blkJobInfo)
 
+            # best-effort attempt to tear down the replica
+            try:
+                self.cif.teardownVolumePath(drive.diskReplicate)
```

Comment:
do we need this on in the block in line 4433 as well?

Reply:
I don't think so since `isDiskReplicationInProgress` checks the existence of the `diskReplicate` attribute - can it be that the replica has been created and this attribute is not set? and if so, how can we get the replica then?
looking at the logs from the environment I used, they seem to align with the log of the reported issue in which we get to the block above after stepping in this block:
```
2023-04-19 08:43:22,693+0200 ERROR (jsonrpc/7) [virt.vm] (vmId='220393e2-bad8-483e-b57c-1ac9fb4dd9c3') Replication job not found (drive: 'sdb', srcDisk: {'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}, job: {}) (vm:4461)
2023-04-19 08:43:22,695+0200 INFO  (jsonrpc/7) [api.virt] FINISH diskReplicateFinish return={'status': {'code': 55, 'message': 'Drive replication error'}} from=::ffff:192.168.178.4,43184, flow_id=993f279b-24b4-4d8f-89b2-e304c7cf14ed, vmId=220393e2-bad8-483e-b57c-1ac9fb4dd9c3 (api:54)
2023-04-19 08:43:22,695+0200 INFO  (jsonrpc/7) [jsonrpc.JsonRpcServer] RPC call VM.diskReplicateFinish failed (error 55) in 0.00 seconds (__init__:312)
2023-04-19 08:43:23,730+0200 INFO  (jsonrpc/2) [api.virt] START diskReplicateFinish(srcDisk={'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}, dstDisk={'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}) from=::ffff:192.168.178.4,43184, flow_id=993f279b-24b4-4d8f-89b2-e304c7cf14ed, vmId=220393e2-bad8-483e-b57c-1ac9fb4dd9c3 (api:48)
2023-04-19 08:43:23,731+0200 ERROR (jsonrpc/2) [api] FINISH diskReplicateFinish error=Replication not in progress.: {'vmId': '220393e2-bad8-483e-b57c-1ac9fb4dd9c3', 'driveName': 'sdb', 'srcDisk': {'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}} (api:131)
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/vdsm/common/api.py", line 124, in method
    ret = func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/vdsm/API.py", line 558, in diskReplicateFinish
    return self.vm.diskReplicateFinish(srcDisk, dstDisk)
  File "/usr/lib/python3.6/site-packages/vdsm/virt/vm.py", line 4452, in diskReplicateFinish
    srcDisk=srcDisk)
vdsm.common.exception.ReplicationNotInProgress: Replication not in progress.: {'vmId': '220393e2-bad8-483e-b57c-1ac9fb4dd9c3', 'driveName': 'sdb', 'srcDisk': {'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}}
2023-04-19 08:43:23,733+0200 INFO  (jsonrpc/2) [api.virt] FINISH diskReplicateFinish return={'status': {'code': 88, 'message': "Replication not in progress.: {'vmId': '220393e2-bad8-483e-b57c-1ac9fb4dd9c3', 'driveName': 'sdb', 'srcDisk': {'imageID': '13d70754-e313-4284-afd7-e817b4fe4f99', 'poolID': '77b3bd70-0c5e-11ea-b02c-52540090e3ee', 'volumeID': '3ad1f9de-8b1d-42a1-9f11-9d137ba709f1', 'device': 'disk', 'domainID': '22c8f6ba-3beb-4c8a-9012-41508cb96db2'}}"}} from=::ffff:192.168.178.4,43184, flow_id=993f279b-24b4-4d8f-89b2-e304c7cf14ed, vmId=220393e2-bad8-483e-b57c-1ac9fb4dd9c3 (api:54)
```